packages/application-generic/src/services/in-memory-provider/in-memory-provider.service.ts

```diff
@@ -80,12 +86,21 @@ export class InMemoryProviderService {
     return this.getStatus() === CLIENT_READY;
   }
 
-  public isClusterMode(): boolean {
-    return process.env.IN_MEMORY_CLUSTER_MODE_ENABLED === 'true';
+  public async isClusterMode(): Promise<boolean> {
+    const isClusterModeEnabled =
```

Comment:
Now we can rule this by the feature flags service. For our self hosted users that would mean that if they do not plug a feature flags service they will fallback to the environment variable set (in the default .env files provided would be `false`) and if not setting the environment variable files, would fallback to `false` as the hardcoded value.